tools/clang/unittests/HLSL/DxilContainerTest.cpp

```diff
@@ -1762,6 +1763,162 @@ TEST_F(DxilContainerTest, DxcUtils_CreateReflection) {
   }
 }
 
+TEST_F(DxilContainerTest, CheckReflectionQueryInterface) {
+  if (m_ver.SkipDxilVersion(1, 5))
+    return;
+
+  // Check that QueryInterface for shader and library reflection accepts/rejects
+  // interfaces properly
+
+  CComPtr<IDxcUtils> pUtils;
+  VERIFY_SUCCEEDED(m_dllSupport.CreateInstance(CLSID_DxcUtils, &pUtils));
+
+  CComPtr<IDxcCompiler> pCompiler;
+  CComPtr<IDxcBlobEncoding> pSource;
+  VERIFY_SUCCEEDED(CreateCompiler(&pCompiler));
+  CreateBlobFromText(Ref1_Shader, &pSource);
+
+  // ID3D12LibraryReflection
+  {
+    CComPtr<IDxcOperationResult> pResult;
+    CComPtr<IDxcResult> pResultV2;
+    CComPtr<IDxcBlob> pReflectionPart;
+    VERIFY_SUCCEEDED(pCompiler->Compile(pSource, L"hlsl.hlsl", L"", L"lib_6_3",
+                                        nullptr, 0, nullptr, 0, nullptr,
+                                        &pResult));
+    VERIFY_SUCCEEDED(pResult->QueryInterface(&pResultV2));
+    VERIFY_SUCCEEDED(pResultV2->GetOutput(
+        DXC_OUT_REFLECTION, IID_PPV_ARGS(&pReflectionPart), nullptr));
+    DxcBuffer buffer = {pReflectionPart->GetBufferPointer(),
+                        pReflectionPart->GetBufferSize(), 0};
+
+    {
+      CComPtr<ID3D12LibraryReflection> pLibraryReflection;
+      VERIFY_SUCCEEDED(
+          pUtils->CreateReflection(&buffer, IID_PPV_ARGS(&pLibraryReflection)));
+
+      CComPtr<ID3D12LibraryReflection> pLibraryReflection2;
+      VERIFY_SUCCEEDED(pLibraryReflection->QueryInterface(IID_PPV_ARGS(&pLibraryReflection2)));
```

Comment:
Was this really failing before your change? I think if not, it can be removed, this seems like very basic functionality.

Reply:
Made a change to `CreateDxilLibraryReflection`, might as well test some basic QueryInterface while we're at it for completeness.